src/tiled/changemapobject.h

```diff
@@ -111,5 +111,26 @@ class ChangeMapObjectsTile : public QUndoCommand
     QVector<bool> mOldChangeStates;
 };
 
+class DetachObject : public QUndoCommand
+{
+public:
+    /**
+     * Creates an undo command that detaches the given template instance from the template.
+     */
+    DetachObject(MapDocument *mapDocument,
+                 MapObject *mapObject);
```

Comment:
For efficiency reasons, please make it a `DetachObjects` undo command that takes a list of objects.

Reply:
Okay, I think bulking the commands together would be more efficient. I did it on individual objects to avoid using lists inside the undo command.